api/src/test/java/org/apache/iceberg/types/TestConversions.java

```diff
@@ -111,9 +111,9 @@ public void testByteBufferConversions() {
     assertConversion(
         400000000L, TimestampNanoType.withZone(), new byte[] {0, -124, -41, 23, 0, 0, 0, 0});
     assertThat(Literal.of(400000L).to(TimestampNanoType.withoutZone()).toByteBuffer().array())
-        .isEqualTo(new byte[] {0, -124, -41, 23, 0, 0, 0, 0});
+        .isEqualTo(new byte[] {-128, 26, 6, 0, 0, 0, 0, 0});
     assertThat(Literal.of(400000L).to(TimestampNanoType.withZone()).toByteBuffer().array())
```

Comment:
I'm a bit confused how the original assertion was passing? Shouldn't have this always been equivalent to `{-128, 26, 6, 0, 0, 0, 0, 0}`?

Reply:
I think the cause is the original logic called `DateTimeUtil.microsToNanos` method which multiples the value by 1000:
https://github.com/apache/iceberg/blob/b9b61b1d72ebb192d5e90453ff7030ece73d2603/api/src/main/java/org/apache/iceberg/expressions/Literals.java#L444-L445

Reply:
Ah I see the comment on line 107/108. Could we update the `assertConversion` to instead test against 400000L and then remove the comment. At this point we are no longer having to pass in different values since we Literal.of(someLong).to(TimestampNanos) will always interpret someLong as nanoseconds.